### PR TITLE
KAFKA-16914: Added share group dynamic and broker configs

### PR DESCRIPTION
**Pull request by adixitconfluent**
### About
KIP-932 introduces a bunch of broker and dynamic configs for share groups. This PR adds those new configs. The changes include:

1. Defined `ShareGroupConfigs` class which stores various share group configurations.
2. Use the defined share configs in `KafkaConfig.scala` for making it available to `BrokerServer`
3. Adds a few tests to validate the conditions on these new configs.

### CODE REVIEW

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 1]
```diff
@@ -1455,6 +1487,30 @@ class KafkaConfig private(doLog: Boolean, val props: java.util.Map[_, _], dynami
...
+      s"${ShareGroupConfigs.SHARE_GROUP_MAX_SESSION_TIMEOUT_MS_CONFIG} must be greater than or equals " +
+        s"to ${ShareGroupConfigs.SHARE_GROUP_MIN_SESSION_TIMEOUT_MS_CONFIG}")
+    require(shareGroupSessionTimeoutMs >= shareGroupMinSessionTimeoutMs,
+      s"${ShareGroupConfigs.SHARE_GROUP_SESSION_TIMEOUT_MS_CONFIG} must be greater than or equals " +
+        s"to ${ShareGroupConfigs.SHARE_GROUP_MIN_SESSION_TIMEOUT_MS_CONFIG}")
+    require(shareGroupSessionTimeoutMs <= shareGroupMaxSessionTimeoutMs,
+      s"${ShareGroupConfigs.SHARE_GROUP_SESSION_TIMEOUT_MS_CONFIG} must be less than or equals " +
+        s"to ${ShareGroupConfigs.SHARE_GROUP_MAX_SESSION_TIMEOUT_MS_CONFIG}")
+
+    require(shareGroupMaxRecordLockDurationMs >= shareGroupRecordLockDurationMs,
```
Comment (omkreddy) [inline core/src/main/scala/kafka/server/KafkaConfig.scala:1517]: Dont we need to add check for `SHARE_GROUP_MIN_RECORD_LOCK_DURATION_MS_CONFIG`  similar to above checks for `shareGroupMaxSessionTimeoutMs`

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 2]
```diff
@@ -292,6 +292,23 @@ object KafkaConfig {
...
       .define(GroupCoordinatorConfig.CONSUMER_GROUP_MIGRATION_POLICY_CONFIG, STRING, GroupCoordinatorConfig.CONSUMER_GROUP_MIGRATION_POLICY_DEFAULT, CaseInsensitiveValidString.in(Utils.enumOptions(classOf[ConsumerGroupMigrationPolicy]): _*), MEDIUM, GroupCoordinatorConfig.CONSUMER_GROUP_MIGRATION_POLICY_DOC)
 
+      /** Share Group Configurations **/
+      // Internal configuration used by integration and system tests.
+      .defineInternal(ShareGroupConfigs.SHARE_GROUP_ENABLE_CONFIG, BOOLEAN, ShareGroupConfigs.SHARE_GROUP_ENABLE_DEFAULT, null, MEDIUM, ShareGroupConfigs.SHARE_GROUP_ENABLE_DOC)
+      .define(ShareGroupConfigs.SHARE_GROUP_DELIVERY_COUNT_LIMIT_CONFIG, INT, ShareGroupConfigs.SHARE_GROUP_DELIVERY_COUNT_LIMIT_DEFAULT, between(2, 10), MEDIUM, ShareGroupConfigs.SHARE_GROUP_DELIVERY_COUNT_LIMIT_DOC)
+      .define(ShareGroupConfigs.SHARE_GROUP_RECORD_LOCK_DURATION_MS_CONFIG, INT, ShareGroupConfigs.SHARE_GROUP_RECORD_LOCK_DURATION_MS_DEFAULT, between(1000, 60000), MEDIUM, ShareGroupConfigs.SHARE_GROUP_RECORD_LOCK_DURATION_MS_DOC)
```
Comment (omkreddy) [inline core/src/main/scala/kafka/server/KafkaConfig.scala]: `between(1000, 60000)` -This should be `atleast(` validator and add require check below validation checks?

File: core/src/main/scala/kafka/server/KafkaConfig.scala (hunk 1 above)
Comment (adixitconfluent) [reply core/src/main/scala/kafka/server/KafkaConfig.scala:1517]: sure, I'll add it

**Review by omkreddy — APPROVED** [review]
@adixitconfluent Thanks for the PR. LGTM

**Pull request merged**